src/app/Http/Controllers/CrudController.php

```diff
@@ -118,12 +119,15 @@ public function storeCrud(StoreRequest $request = null)
         $this->data['entry'] = $this->crud->entry = $item;
 
         // show a success message
-        \Alert::success(trans('backpack::crud.insert_success'))->flash();
+        $message = trans('backpack::crud.insert_success');
+        \Alert::success($message)->flash();
 
         // save the redirect choice for next time
         $this->setSaveAction();
 
-        return $this->performSaveAction($item->getKey());
+        $redirectUrl = $this->performSaveAction($item->getKey());
+
+        return \Request::instance()->ajax() ? response()->json(['redirect_url' => $redirectUrl->getTargetUrl(), 'message' => $message]) : \Redirect::to($redirectUrl);
```

Comment:
Isn't `$request` already here?

Also can we remove the ternary here and just use `if {}` type clauses.